src/v/redpanda/admin_server.cc

```diff
@@ -463,6 +463,29 @@ bool get_boolean_query_param(
     return ss::http::request::case_insensitive_cmp()(str_param, "true")
            || str_param == "1";
 }
+
+/**
+ * Helper for requests with decimal_integer URL query parameters.
+ *
+ * Throws a bad_request exception if the parameter is present but not
+ * an integer.
+ */
+std::optional<uint64_t>
+get_integer_query_param(const ss::http::request& req, std::string_view name) {
+    auto key = ss::sstring(name);
+    if (!req.query_parameters.contains(key)) {
+        return std::nullopt;
+    }
+
+    const ss::sstring& str_param = req.query_parameters.at(key);
+    try {
+        return std::stoi(str_param);
```

Comment:
I see that in other places we use boost::lexical_cast. but I have no idea of the tradeoffs

Reply:
Yeah, I don't know that there's any particular benefit to the boost way :shrug: